cognitedata/reveal
# feat: individual model clipping planes

#### Type of change

![Feat](https://img.shields.io/badge/Type-Feat-green)  

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/REV-682

## Description :pencil:


Adds the ability to set clipping planes individually for each model, in addition to globally. The global and "local" clipping planes will both be evaluated, resulting in the intersection of the two volumes.

In addition:
- Deprecated `set/getClippingPlanes` in the viewer, added `set/getGlobalClippingPlanes` instead
- Disable picking for clipped points in point clouds
- Expanded clipping plane UI

## How has this been tested? :mag:



Tested our local models (both CAD and pointcloud) and see that clipping planes smoothly interact when set from the UI.

Point cloud picking: 
- Open https://localhost:3000/?project=3d-test&env=greenfield-3d&modelId=7394378315995345&revisionId=36918959705705
- Wait for object load, click fit to camera when done
- Move the camera so that one of the big towers are hidden behind another
- Clip the closest with clipping planes, ensuring the one behind is now visible
- Click it and see that it's highlighted

## Test instructions :information_source:




## Checklist :ballot_box_with_check:
